src/commands.rs

```diff
@@ -475,7 +485,16 @@ where
                         }
                         _ => {
                             //TODO: something better here?
-                            return Err(e).context("error reading compile response from server");
+                            eprintln!("unexpected io error: {:?}", e.downcast_ref::<io::Error>());
+                            if ignore_all_server_io_errors() {
+                                eprintln!(
+                                    "sccache: warning: error reading compile response from server \
+                                     compiling locally instead (option SCCACHE_IGNORE_SERVER_IO_ERROR passed)"
```

Comment:
IMO, the "option SCCACHE_IGNORE_SERVER_IO_ERROR passed" part of this message is redundant, let's remove it.
Worst comes to worst, if someone is _really_ curious why they're seeing this error message, they can check the source code.
This'll also make use consistent with the "server looks like it shut down" message.
```suggestion
                                    "sccache: warning: error reading compile response from server, \
                                     compiling locally instead"
```

Reply:
Agree! revert the commit https://github.com/mozilla/sccache/pull/1090/commits/878988e4ebe8a7bc99393f5355da6518a5909edc

Reply:
Whoops, missed that Sylvestre requested that, my mistake. I think it's ok either way, though I have my preference ;)